rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php

```diff
@@ -116,30 +117,30 @@ public function refactor(Node $node): ?Node
                 continue;
             }
 
+            $props = $property->props[0];
+            $propertyName = $this->getName($props);
             $this->removeNode($property);
             $this->removeNode($promotionCandidate->getAssign());
 
-            $property = $promotionCandidate->getProperty();
+            /** @var string $paramName */
             $paramName = $this->getName($param);
 
-            // rename also following calls
-            $propertyName = $this->getName($property->props[0]);
-            /** @var string $oldName */
-            $oldName = $this->getName($param->var);
-            $this->variableRenamer->renameVariableInFunctionLike($constructClassMethod, $oldName, $propertyName, null);
+            $this->variableRenamer->renameVariableInFunctionLike($constructClassMethod, $propertyName, $paramName, null);
+
+            $propertyFetches = $this->propertyFetchFinder->findLocalPropertyFetchesByName($node, $propertyName);
+            $newName = new Identifier($paramName);
+            foreach ($propertyFetches as $propertyFetch) {
+                $propertyFetch->name = $newName;
+            }
 
             $paramTagValueNode = $classMethodPhpDocInfo->getParamTagValueNodeByName($paramName);
 
             if (! $paramTagValueNode instanceof ParamTagValueNode) {
                 $this->decorateParamWithPropertyPhpDocInfo($constructClassMethod, $property, $param, $paramName);
-            } elseif ($paramTagValueNode->parameterName !== '$' . $propertyName) {
-                $paramTagValueNode->parameterName = '$' . $propertyName;
-                $paramTagValueNode->setAttribute(PhpDocAttributeKey::ORIG_NODE, null);
             }
 
-            // property name has higher priority
-            $propertyName = $this->getName($property);
-            $param->var->name = $propertyName;
+            // param name has higher priority
```

Comment:
this will BC break for public and protected modifier, also, even private may indirectly fill parent private class, eg: 

```php
<?php

class SomeParent
{
    private $somePrivate;
    
    public function fill($var, $value)
    {
        $this->$var = $value;
        return $this;
    }
    
    public function get()
    {
        return $this->somePrivate;
    }
}

class SomeChild extends SomeParent
{
}

echo (new SomeChild())->fill('somePrivate', 'value')->get();
```

ref https://3v4l.org/EUYDN

Reply:
This will also make a bug when other property name exists by rely on parameter name, that will make invalid fill property value, and collision, eg: both property `$id` and  `$ID` exists